### ISSUE TITLE
Improvement for comparision against fn

### ISSUE THREAD

**Issue by tumdum** [rust-lang/rust#59488]
```rust
fn foo() -> i32 {
    42
}

fn main() {
    foo > 12;
}
```

([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=94b61ff05ac39b98230b8b99fe7fcfbf))

Errors:

```
   Compiling playground v0.0.1 (/playground)
error[E0369]: binary operation `>` cannot be applied to type `fn() -> i32 {foo}`
 --> src/main.rs:6:5
  |
6 |     foo > 12;
  |     ^^^^^^^^
  |
  = note: an implementation of `std::cmp::PartialOrd` might be missing for `fn() -> i32 {foo}`

error[E0308]: mismatched types
 --> src/main.rs:6:11
  |
6 |     foo > 12;
  |           ^^ expected fn item, found integer
  |
  = note: expected type `fn() -> i32 {foo}`
             found type `{integer}`

error: aborting due to 2 previous errors
```
A much more useful note would be to suggest to the user that a instead of comparing function she should compare result of function evaluation - rustc should suggest that user should add pair of parentheses to call a function ("did you forget the ‘()’ ?").

**Comment by rchaser53** [comment]
I'd like to work on this.
Add the sentence "did you forget the ‘()’ ?" like below.
If is it ok, I will send the PR.
@estebank

```
error[E0369]: binary operation `>` cannot be applied to type `fn() -> i32 {foo}`
  --> $DIR/issue-59488.rs:6:9
   |
LL |     foo > 12;
   |     --- ^ -- {integer}
   |     |
   |     fn() -> i32 {foo}
   |     did you forget `()`?
   |
   = note: an implementation of `std::cmp::PartialOrd` might be missing for `fn() -> i32 {foo}`
```

**Comment by JohnTitor** [comment]
Is this fixed in #59798?

**Comment by rchaser53** [comment]
Yes. #59798 happens degraded. but it will be resolved in #60073

**Comment by davidtwco** [comment]
Closing, suggestion is implemented on nightly:

```
error[E0369]: binary operation `>` cannot be applied to type `fn() -> i32 {foo}`
 --> src/main.rs:6:9
  |
6 |     foo > 12;
  |     --- ^ -- {integer}
  |     |
  |     fn() -> i32 {foo}
  |     help: you might have forgotten to call this function: `foo()`
```
